src/som/interpreter/Interpreter.java

```diff
@@ -50,176 +52,211 @@ public Interpreter(final Universe universe) {
     this.universe = universe;
   }
 
-  private void doDup(final Frame frame) {
-    // Handle the DUP bytecode
-    frame.push(frame.getStackElement(0));
+  private void doDup(final VirtualFrame frame)
+      throws FrameSlotTypeException {
+    SAbstractObject value = StackUtils.getRelativeStackElement(frame, 0);
```

Comment:
What does the "relative" refer to here? Relative to what?

Reply:
Not a good naming I admit. That's to disambiguate with `getStackElement(VirtualFrame frame, int index)` which returns the element at the given index.
The `getRelativeStackElement` returns the top of the stack if the provided index is equal to 0. So, in most cases in the project, it is indeed a `peek` but not always.